plutus-ledger-api/src/PlutusLedgerApi/Common/Eval.hs

```diff
@@ -53,6 +54,7 @@ data EvaluationError =
     | CodecError !ScriptDecodeError -- ^ A deserialisation error
     -- TODO: make this error more informative when we have more information about what went wrong
     | CostModelParameterMismatch -- ^ An error indicating that the cost model parameters didn't match what we expected
+    | NonUnitReturnValue -- ^ The script evaluated to a value that is not the builtin unit.
```

Comment:
Maybe a more descript name like `NonUnitReturnValueConway`, to remind ourselves that this can only happen with Conway onwards

Reply:
It's not about Conway but about V3. I changed the name to `InvalidReturnValue`, and made the `Pretty` instance more descriptive.